code/game/turfs/open/floor/prison_floor.dm

```diff
@@ -0,0 +1,97 @@
+
+/turf/open/floor/prison
+	name = "secure floor"
+	desc = "Prison break-proof!"
+	icon = 'icons/turf/prisonfloor.dmi'
+	icon_state = "prisonfloor4"
+	max_integrity = 500
+	holodeck_compatible = TRUE
+	thermal_conductivity = 0.025
+	heat_capacity = INFINITY
+	floor_tile = /obj/item/stack/sheet/plasteel
+	footstep = FOOTSTEP_PLATING
+	barefootstep = FOOTSTEP_HARD_BAREFOOT
+	clawfootstep = FOOTSTEP_HARD_CLAW
+	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	tiled_dirt = FALSE
+	var/plates_type = /obj/item/stack/tile/plasteel
+	var/plates = 4
+	var/wrenching = FALSE
+
+/turf/open/floor/prison/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The reinforcement plates are <b>wrenched</b> firmly in place.</span>"
+
+/turf/open/floor/prison/break_tile()
+	return //unbreakable
+
+/turf/open/floor/prison/burn_tile() // consider changing this
+	return //unburnable
+
+/turf/open/floor/prison/make_plating(force = 0) // I don't know what that is
+	if(force)
+		..()
+	return //unplateable
```

Comment:
This function converts the floor into plating for whne it is destroyed. Normal plasteel tiles call ScrapeAway which just destroys the tile and reveals the tile underneath.
This is called by singularities.